src/constants/contracts/contracts.ts

```diff
@@ -1,8 +1,18 @@
-import { type ContractConfig } from '@wagmi/cli'
 import { type Abi, Address, erc20Abi, isAddress } from 'viem'
+import { mainnet, sepolia } from 'viem/chains'
 
 import { ENSRegistryABI } from '@/src/constants/contracts/abis/ENSRegistry'
-import { type ChainsIds, type RequiredChainId } from '@/src/lib/networks.config'
+import { type ChainsIds } from '@/src/lib/networks.config'
+
+type ValidateId<T> = T extends ChainsIds ? T : 'Invalid ID – This ID is not permitted'
+type RequiredChainIds = ValidateId<typeof mainnet.id> // this can be extended ValidateId<typeof mainnet.id | sepolia.id | ...>
+type ContractConfigAddress = Record<RequiredChainIds, string> & Partial<Record<ChainsIds, string>>
```

Comment:
Not sure about what this does. 
```suggestion
type RequiredAddresses = Record<RequiredChainIds, Address> 
type OptionalAddresses = Partial<Record<ChainsIds, Address>>

type ContractConfigAddress = RequiredAddresses & OptionalAddresses 
```

Reply:
But maybe can be rewritten this way